libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -0,0 +1,58 @@
+package com.mapbox.navigation.metrics
+
+import android.content.Context
+import com.google.gson.Gson
+import com.mapbox.android.telemetry.AppUserTurnstile
+import com.mapbox.android.telemetry.MapboxTelemetry
+import com.mapbox.navigation.base.metrics.MetricEvent
+import com.mapbox.navigation.base.metrics.MetricsObserver
+import com.mapbox.navigation.base.metrics.MetricsReporter
+import com.mapbox.navigation.base.metrics.NavigationMetrics
+import com.mapbox.navigation.utils.thread.WorkThreadHandler
+
+object MapboxMetricsReporter : MetricsReporter {
+
+    private val gson = Gson()
+    private lateinit var mapboxTelemetry: MapboxTelemetry
+    @Volatile
+    private var metricsObserver: MetricsObserver? = null
+    private val threadWorker = WorkThreadHandler("MapboxMetricsReporter")
+
+    @JvmStatic
+    fun init(
+        context: Context,
+        accessToken: String,
+        userAgent: String
+    ) {
+        mapboxTelemetry = MapboxTelemetry(context, accessToken, userAgent)
+        mapboxTelemetry.enable()
+    }
+
+    @JvmStatic
+    fun disable() {
+        mapboxTelemetry.disable()
+    }
+
+    // TODO: Do we need this event?
+    fun addAppTurnstileEvent(event: AppUserTurnstile) {
+        mapboxTelemetry.push(event)
+
+        threadWorker.post {
+            metricsObserver?.onMetricUpdated(NavigationMetrics.APP_USER_TURNSTILE, gson.toJson(event))
```

Comment:
Do we want to deliver the events on a worker thread, or should we just do the conversion on a worker and deliver on main?

Reply:
Yes, we want to deliver also on a worker thread, because we don't know what our customers will do under the hood. For example, if they will write events to file storage or send it to network and so on. I mean, we need to protect us (and them) from perform long-term operations in main thread.

Reply:
I don't think we should tbh. If developers are expecting to perform disk operation/networking, they should and probably would anyway maintain concurrency themselves. It's a rather uncommon pattern for an Android library to return on a worker thread and can definitely come as a surprise. Especially since we do not fetch any data, calling observer on the main thread would be preferable in my opinion. Wdyt @mapbox/navigation-android?

Reply:
i'd in general agree another option would be to dispatch callback on which it was registered. that said telem calls may potentially impact perf of our core features which could be hard to debug into. i'd suggest to roll with the current solution in this PR and rethink it as we refactor telem in new modules